consensus/src/chained_bft/sync_manager.rs

```diff
@@ -203,9 +204,13 @@ where
         {
             return Ok(());
         }
+        let peer_str = match peer {
+            Some(p) => p.short_str(),
+            None => "[no preferred peer]".to_string(),
+        };
         debug!(
             "Start state sync with peer: {}, to block: {}, round: {} from {}",
-            peer.short_str(),
```

Comment:
```peer.map_or_else(|x|x.short_str(), ||"[no preferred peer]".to_string())```
It's generally better to have those things inside log statements, so they don't execute when log macros turned off